src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,57 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator
+{
+    /*
+    쉼표(,) 또는 콜론(:)을 구분자로 가지는 문자열을 전달하는 경우 구분자를 기준으로 분리한 각 숫자의 합을 반환
+    (예: “” => 0, "1,2" => 3, "1,2,3" => 6, “1,2:3” => 6)
+    앞의 기본 구분자(쉼표, 콜론)외에 커스텀 구분자를 지정할 수 있다.
+    커스텀 구분자는 문자열 앞부분의 “//”와 “\n” 사이에 위치하는 문자를 커스텀 구분자로 사용한다.
+    예를 들어 “//;\n1;2;3”과 같이 값을 입력할 경우 커스텀 구분자는 세미콜론(;)이며, 결과 값은 6이 반환되어야 한다.
+    문자열 계산기에 숫자 이외의 값 또는 음수를 전달하는 경우 RuntimeException 예외를 throw한다.
+     */
+    private static String DELIMITER = ",|:";
+
+    public int add(String str){
+        return getSum(toIntArr(getStringWhenHasDelimiter(getZeroWhenBlank(str)).split(DELIMITER)));
+    }
+
+    private int[] toIntArr(String[] strArr) {
+        int[] intArr = new int[strArr.length];
+        for(int i = 0; i < strArr.length; i++){
+            intArr[i] = toInt(strArr[i]);
+        }
+        return intArr;
+    }
+
+    private int toInt(String s) {
+        int num = Integer.parseInt(s);
+        if(num < 0){
+            throw new RuntimeException("음수는 허용하지 않습니다.");
```

Comment:
지금 메시지도 좋지만 어떤 값이 예외를 발생시켰는지 메시지에 포함하면 더 좋을 것 같다는 생각이 드네요
그러면 디버깅할 때 더 편리할 것 같아요

또한, RuntimeException 대신 언어에서 제공하는 IllegalArgumentException 같은 표준 예외를 사용하면 잘못된 인자가 전달되었다는 걸 명확히 표현할 수 있어서 더 적절할 것 같아요!

```suggestion
            throw new IllegalArgumentException(num + " 음수는 허용하지 않습니다.");
```